File: sunflower-core/src/main/java/org/tdf/sunflower/controller/TransactionReceiptDTOExt.java
package org.tdf.sunflower.controller;

import org.tdf.sunflower.types.TransactionInfo;
import org.tdf.sunflower.types.Block;

import static org.tdf.sunflower.controller.TypeConverter.toJsonHex;

public class TransactionReceiptDTOExt extends TransactionReceiptDTO {

    public String returnData;
    public String error;

    public TransactionReceiptDTOExt(Block block, TransactionInfo txInfo) {
        super(block, txInfo);
        returnData = toJsonHex(txInfo.getReceipt().getExecutionResult());
        error = txInfo.getReceipt().getError();
    }
}

File: sunflower-core/src/main/java/org/tdf/sunflower/controller/TransactionReceiptDTOExt.kt
package org.tdf.sunflower.controller

import org.tdf.sunflower.types.Block
import org.tdf.sunflower.types.TransactionInfo

class TransactionReceiptDTOExt(block: Block?, txInfo: TransactionInfo) : TransactionReceiptDTO(block, txInfo) {
    var returnData: String
    var error: String

    init {
        returnData = txInfo.receipt.executionResult.jsonHex
        error = txInfo.receipt.error
    }
}